swap_meet/vendor.py

```diff
@@ -1,2 +1,233 @@
+
+    swap_first_item(vendor):
+        swaps first items from my list and param vendor's list 
+        Returns true if both lists are not empty
+
+    return_best_by_category(category):
+        returns item with best condition in vendor list given the category
+
+    swap_best_by_category(other, my_priority, their_priority):
+        checks for your best item given other's priority
+        checks for their best item given your priority
+        swaps those items
+        returns true if swap occurs
+    
+    get_newest():
+        returns item in self.inventory with the lowest age attribute value
+
+    swap_by_newest(other):
+        swaps your's and other vendor's newest items
+        returns true if swap is successful
+    """
+
+
+    def __init__(self, inventory = None): #do not set keyword argument to list bc it is a mutable type
+        """
+        Constructs all the necessary attributes for the vendor object.
+
+        Parameters
+        ----------
+            inventory : list
+                list of strings representing items in inventory
+        """
+
+        if not inventory:
+            self.inventory = []
+        else:
+            self.inventory = inventory
+
+
+    def add(self, item):
+        """
+        Appends the item to the inventory list.
+
+        Parameters
+        ----------
+        item : object
+
+        Returns
+        -------
+        item
+        """
+        self.inventory.append(item)
+        return item
+
+    
+    def remove(self, item):
+        """
+        Removes item from inventory list.
+
+        Parameters
+        ----------
+        item : str
+
+        Returns
+        -------
+        item or None
+        """
+        try:
+            self.inventory.remove(item)
+            return item
+        except ValueError:
+            return False
+    
+
+    def get_by_category(self, category):
+        """
+        Creates a list of items from inventory whose category attributes match the input category. 
+
+        Parameters
+        ----------
+        category : str
+
+        Returns
+        -------
+        items: list
+        """
+
+        return [item for item in self.inventory if item.category == category]
+    
+
+    def swap_items(self, vendor, my_item, vendor_item):
+        """
+        Removes and appends 2nd and 3rd parameters from my inventory respectively. Appends and removes 2nd and 3rd parameters from vendor's inventory respectively.
+
+        Parameters
+        -------
+        vendor: composite object
+        my item: component object 
+        vendor_item: component object
+
+        Returns
+        -------
+        self.inventory: object
+        
+        """
+        if my_item in self.inventory and vendor_item in vendor.inventory:
+                self.remove(my_item) 
+                self.add(vendor_item)            
+                vendor.remove(vendor_item)       
+                vendor.add(my_item) 
+
+                return True
```

Comment:
To unindent the primary focus of this function, we can reverse the conditional and use a guard clause. 

```python
if my_item not in self.inventory and vendor_item not in vendor.inventory:
    return False
    
self.remove(my_item) 
self.add(vendor_item)            
vendor.remove(vendor_item)       
vendor.add(my_item)

return True
```